src/awkward/_v2/contents/content.py

```diff
@@ -1406,12 +1408,30 @@ def to_list(self, behavior=None):
         return self.packed()._to_list(behavior, None)
 
     def _to_list_custom(self, behavior, json_conversions):
-        cls = ak._v2._util.arrayclass(self, behavior)
-        if cls.__getitem__ is not ak._v2.highlevel.Array.__getitem__:
-            array = cls(self)
+        if self.is_RecordType:
+            getitem = ak._v2._util.recordclass(self, behavior).__getitem__
+            overloaded = (
+                getitem is not ak._v2.highlevel.Record.__getitem__
+                and not getattr(getitem, "ignore_in_to_list", False)
```

Comment:
If we decide to use a flag, then I'd suggest also adding a decorator to our high level API set it, e.g.

```python
def ignore_in_to_list(func):
    func.ignore_in_to_list = True
    return func
```

Reply:
I guess this could go in highlevel.py (not top-level in `ak.*`):

```python
class Override(ak._v2.Array):
    @ak._v2.highlevel.ignore_in_to_list
    def __getitem__(self, where):
        return super().__getitem__(where)
```

While being public API, it's only going to be of interest to a few people.